include/aws/common/math.cbmc.inl

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+/*
+ * This header is already included, but include it again to make editor
+ * highlighting happier.
+ */
+#include <aws/common/common.h>
+
+/* This header does safe operations - supressing the checks within these functions
+ * avoids unnecessary CBMC assertions
+ */
+#pragma CPROVER check push
+#pragma CPROVER check disable "unsigned-overflow"
```

Comment:
Why do we need to disable the `unsigned-overflow-check` here if the functions perform overflow checks before each arithmetic operation?

Reply:
Performance. Don't need to recheck the same property twice.